### DISCUSSION TITLE
[RRFC] delete node_modules folder on `npm ci` failure

### DISCUSSION THREAD

**Discussion by isaacs** [npm/rfcs#79] [General]
Right now `npm ci` doesn't do any cleanup when it fails.  This is hazardous because the node_modules folder might be in a broken state, but since `npm ci` also doesn't do any staging step, the broken packages are in place and require()-able.

Since `npm ci` always starts from (or creates) a clean slate, it should remove the `node_modules` folder if the installation fails.

**Comment by ChALkeR** [comment]
Thanks for the update!

**Comment by ruyadorno** [comment]
@claudiahdz is this still the case in current `v7.0.0-beta` branch?
